### PR TITLE
Cleanup of sourceforge download links

### PR DESCRIPTION
**Pull request by phanimahesh**
This is a work-in-progress clean up effort.
The commits will be broken down into one per derivation. Bulk of the effort has been automated with

```
sed -e '/.*url =/ {s=https\?://.*sourceforge.net/=mirror://sourceforge/=}' -i pkgs/**/*.nix
```

I'll mark the files below done once I'm done reviewing them. Help would be most welcome.

Edit: The substitution I used was very naive, and there are a lot of unwanted changes. Broadly,
- svn, cvs links. I had no idea people were still using svn and cvs.
- Links that were on `<project>.sourceforge.net`. I'll have to spend time figuring out the equivalents on downloads.sourceforge.net, or leave them be if that isn't possible.
- [x] pkgs/applications/backup/areca/default.nix
- [ ] pkgs/applications/editors/emacs-modes/bbdb/default.nix
- [ ] pkgs/applications/editors/emacs-modes/jdee/default.nix
- [x] pkgs/applications/editors/ht/default.nix
- [ ] pkgs/applications/graphics/potrace/default.nix
- [ ] pkgs/applications/misc/abook/default.nix
- [x] pkgs/applications/misc/gammu/default.nix
- [x] pkgs/applications/misc/mlterm/default.nix
- [ ] pkgs/applications/misc/sweethome3d/default.nix
- [x] pkgs/applications/misc/xautoclick/default.nix
- [ ] pkgs/applications/office/mmex/default.nix
- [ ] pkgs/applications/science/math/yacas/default.nix
- [ ] pkgs/applications/video/recordmydesktop/default.nix
- [ ] pkgs/applications/video/recordmydesktop/gtk.nix
- [ ] pkgs/applications/video/recordmydesktop/qt.nix
- [x] pkgs/applications/window-managers/windowmaker/dockapps/wmsystemtray.nix
- [ ] pkgs/data/fonts/corefonts/default.nix
- [x] pkgs/desktops/lxde/core/lxmenu-data.nix
- [x] pkgs/development/compilers/fpc/binary.nix
- [x] pkgs/development/compilers/mlton/default.nix
- [x] pkgs/development/compilers/mozart/binary.nix
- [ ] pkgs/development/libraries/armadillo/default.nix
- [ ] pkgs/development/libraries/irrlicht/irrlicht3843.nix
- [ ] pkgs/development/libraries/libgringotts/default.nix
- [ ] pkgs/development/libraries/libmpeg2/default.nix
- [ ] pkgs/development/libraries/plib/default.nix
- [ ] pkgs/development/tools/misc/ctags/default.nix
- [ ] pkgs/development/tools/misc/dfu-util/default.nix
- [x] pkgs/development/tools/omniorb/default.nix
- [ ] pkgs/development/tools/sqsh/default.nix
- [ ] pkgs/games/blobby/default.nix
- [ ] pkgs/games/privateer/default.nix
- [ ] pkgs/games/typespeed/default.nix
- [ ] pkgs/os-specific/linux/procps/watch.nix
- [ ] pkgs/servers/dict/dictd-db.nix
- [x] pkgs/tools/filesystems/fuseiso/default.nix
- [ ] pkgs/tools/filesystems/jfsutils/default.nix
- [ ] pkgs/tools/filesystems/nilfs-utils/default.nix
- [ ] pkgs/tools/misc/cutecom/default.nix
- [ ] pkgs/tools/misc/fondu/default.nix
- [x] pkgs/tools/networking/srelay/default.nix
- [ ] pkgs/tools/system/foremost/default.nix
- [ ] pkgs/tools/system/smartmontools/default.nix
- [ ] pkgs/tools/text/poedit/default.nix
- [ ] pkgs/top-level/python-packages.nix

Ref: #16900

### CODE REVIEW

File: pkgs/applications/editors/emacs-modes/jdee/default.nix [hunk 1]
```diff
@@ -9,7 +9,7 @@ in
     # Last release is too old, so use SVN.
     # See http://www.emacswiki.org/emacs/JavaDevelopmentEnvironment .
     src = fetchsvn {
-      url = "https://jdee.svn.sourceforge.net/svnroot/jdee/trunk/jdee";
+      url = "mirror://sourceforge/svnroot/jdee/trunk/jdee";
```
Comment (phanimahesh) [inline pkgs/applications/editors/emacs-modes/jdee/default.nix]: I don't think this'll work. What shall we do with svn links?

Comment (FRidh) [inline pkgs/applications/editors/emacs-modes/jdee/default.nix]: How old is the release? Is the package still used?

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -3179,7 +3179,7 @@ in modules // {
     disabled = !isPy27;
 
     src = pkgs.fetchurl {
-      url = "http://cddb-py.sourceforge.net/${name}.tar.gz";
+      url = "mirror://sourceforge/${name}.tar.gz";
```
Comment (phanimahesh) [inline pkgs/top-level/python-packages.nix]: http://downloads.sourceforge.net/project/cddb-py/cddb-py/1.4/CDDB-1.4.tar.gz

Add version as separate field and construct url from it.

File: pkgs/applications/misc/sweethome3d/default.nix [hunk 3]
```diff
@@ -82,11 +82,11 @@ in rec {
     desktopName = "Sweet Home 3D";
     icons = {
       "32x32" = fetchurl {
-        url = "http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon32x32.png";
+        url = "mirror://sourceforge/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon32x32.png";
         sha256 = "1r2fhfg27mx00nfv0qj66rhf719s2g1vhdis7bdc9mqk9x0mb0ir";
       };
       "48x48" = fetchurl {
-        url = "http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon48x48.png";
+        url = "mirror://sourceforge/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon48x48.png";
```
Comment (jraygauthier) [inline pkgs/applications/misc/sweethome3d/default.nix]: I do not think these translations will work.

Comment (phanimahesh) [inline pkgs/applications/misc/sweethome3d/default.nix]: yes, as I noted, svn and cvs links need to be filtered out. Will do tomorrow.

File: pkgs/applications/editors/emacs-modes/jdee/default.nix (hunk 1 above)
Comment (phanimahesh) [inline pkgs/applications/editors/emacs-modes/jdee/default.nix]: I have no idea. :(

Comment (RamKromberg) [inline pkgs/applications/editors/emacs-modes/jdee/default.nix]: They moved to https://github.com/jdee-emacs/jdee

**Comment by RamKromberg** [comment]
I'm fairly sure it's irrelevant, but many svn and git sourceforge projects use the _sf.net_ domain. e.g. https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/p2p/transgui/default.nix

Anyhow, those should matter but I figured it should be mentioned just the same.

**Comment by phanimahesh** [comment]
I have rebased this and updated it to only include valid changes. I have a stash of pending changes but want to keep this mergeable. Anyone else who would like to help, please feel free to submit PRs against my branch. Please see PR description for a list of files that need to be worked on.

**Comment by RamKromberg** [comment]
@phanimahesh Wouldn't it be better to merge now, and leave the rest for after 16.09?

**Comment by phanimahesh** [comment]
@RamKromberg I've just been discussing the same on IRC. Yes, it can be merged now and the rest can be moved to an issue under TODO to be tackled as and when possible. I've gotten busier and am afraid this will go stale if not completed soon.

Also, the travis build failure seems to be spurious, log size exceeded 4 MB and it was killed.

**Comment by joachifm** [comment]
Yes, let's merge this as is and move the todo stuff to a tracking issue.

**Comment by RamKromberg** [comment]
Thanks!

**Pull request merged**